File: Playwright/playwright-python/Capacitacion-PW-Python/pages/LoginPage.py
# Clase que representa la página de login
class LoginPage:
    # Constructor de la clase
    def __init__(self, page):
        # Asignar la página de Playwright a un atributo de la clase
        self.page = page
        # Definir los selectores de los elementos de la página
        self.username_input = page.get_by_placeholder("Username")
        self.password_input = page.get_by_placeholder("Password")
        self.login_button = page.get_by_role("button", name="Login")
        self.agregar_producto_button = page.get_by_role("button", name="Add to cart").nth(0)
    
    # Método para realizar login    
    def login(self, username, password):
        # Rellenar los campos de usuario y contraseña y hacer clic en el botón de login
        self.username_input.fill(username)
        self.password_input.fill(password)
        self.login_button.click()
        

File: Playwright/playwright-typescript/sauce-demo-ts/tests/Pages/LoginPage.ts
import { type Locator, type Page } from '@playwright/test';

export class LoginPage {
    // Elementos de la página de login
    readonly page: Page; // Referencia a la página
    readonly username: Locator; // Localizador nombre de usuario
    readonly password: Locator; // Localizador contraseña
    readonly loginButton: Locator; // Localizador botón de login

    constructor(page: Page) {
        this.page = page; // Inicializar la referencia a la página
        this.username = page.locator('[data-test="username"]'); // Inicializar el localizador nombre de usuario
        this.password = page.locator('[data-test="password"]'); // Inicializar el localizador contraseña
        this.loginButton = page.locator('[data-test="login-button"]'); // Inicializar el localizador botón de login
    }

    async login(username: string, password: string) {
        await this.username.fill(username); // Ingresar el nombre de usuario con metodo fill
        await this.password.fill(password); // Ingresar la contraseña con metodo fill
    }

    async clickLogin() {
        await this.loginButton.click(); // Hacer click en el botón de login
    }
}
